### PR TITLE
UnitTests:  accept or decline invitation as anonymous 

### LINKED ISSUE #7469: unit test backend function acceptOrDeclineInvitationAsAnonymous

### PR DESCRIPTION
**Pull request by xmano**
Closes #7469

### CODE REVIEW

File: apps/backend-functions/src/fixtures/inviteUsers.ts [hunk 1]
```diff
@@ -19,8 +19,45 @@ export const testFixture = {
...
+    lastName: 'Two',
     email: 'u2@cascade8.com',
     uid: 'uid-user2',
     orgId: 'O001',
   },
+
+  //Consents
+  'consents/O001': {
+    id: 'O001',
+  },
```
Comment (phpgeekfr) [inline apps/backend-functions/src/fixtures/inviteUsers.ts:32]: file is named `inviteUsers` why do we have consent related data ?

File: apps/backend-functions/src/consent.ts [hunk 2]
```diff
@@ -48,7 +48,7 @@ export const createConsent = async (
 
     const consentSnap = await tx.get(db.doc(`consents/${consent.id}`));
     const consentData = consentSnap.data();
-
+    
     if (consentData) {
       consent = _createConsent(consentData);
     }
```
Comment (phpgeekfr) [inline apps/backend-functions/src/consent.ts:54]: can you revert this file ?

File: apps/backend-functions/src/invitation.spec.ts [hunk 3]
```diff
@@ -62,10 +61,9 @@ describe('Invitation backend-function unit-tests', () => {
       };
 
       expect.assertions(1);
-      await expect(async () => {
-        await wrapped(data, context)
-      }).rejects
-        .toThrow('Permission denied: missing org id.');
+      await expect(wrapped(data, {}))
+            .rejects
+            .toThrow('Permission denied: missing auth context.');
```
Comment (phpgeekfr) [inline apps/backend-functions/src/invitation.spec.ts:66]: this test was changed.  you where testing `.toThrow('Permission denied: missing org id.');` and now `.toThrow('Permission denied: missing auth context.');` 

`context` was removed from `wrapped(data, context)` =>  `wrapped(data, {})`  and now 
```
const context = {
        auth: {
          uid: 'uid-c8',
          token: ''
        }
      };
```

 is un-used

File: apps/backend-functions/src/invitation.spec.ts [hunk 4]
```diff
@@ -153,4 +151,140 @@ describe('Invitation backend-function unit-tests', () => {
       );
     });
   });
+
+  describe('\'acceptOrDeclineInvitationAsAnonymous\' tests', () => {
+    it('missing auth context, throws error', async () => {
+      const wrapped = testEnv.wrap(acceptOrDeclineInvitationAsAnonymous);
+
+      //Compose the call to simpleCallable cf with param data
+      const data = {
```
Comment (phpgeekfr) [inline apps/backend-functions/src/invitation.spec.ts:160]: data is not valid do `const data: AnonymousInvitationAction`

File: apps/backend-functions/src/fixtures/inviteUsers.ts [hunk 5]
```diff
@@ -19,8 +19,45 @@ export const testFixture = {
...
+
+  //Invitations
+  'invitations/O001': {
```
Comment (phpgeekfr) [inline apps/backend-functions/src/fixtures/inviteUsers.ts:35]: this one is not used `O001`

File: apps/backend-functions/src/invitation.spec.ts [hunk 6]
```diff
@@ -153,4 +151,140 @@ describe('Invitation backend-function unit-tests', () => {
...
+      };
+
+      expect.assertions(1);
+      await expect(wrapped(data, {}))
+            .rejects
+            .toThrow('Permission denied: missing auth context.');
+
+    });
+
+    it('Non-existent invitation, throws error', async () => {
```
Comment (phpgeekfr) [inline apps/backend-functions/src/invitation.spec.ts:173]: this invitation `I001` does exists. This is the same test as `it('Invitation type is not \'attendEvent\', throws error', async () => {`

File: apps/backend-functions/src/invitation.spec.ts [hunk 7]
```diff
@@ -153,4 +151,140 @@ describe('Invitation backend-function unit-tests', () => {
...
+      const wrapped = testEnv.wrap(acceptOrDeclineInvitationAsAnonymous);
+
+      //Compose the call to simpleCallable cf with param data
+      const data: AnonymousInvitationAction = {
+        email: 'test@cascade8.com',
+        invitationId: 'I001',
+        status: null
```
Comment (phpgeekfr) [inline apps/backend-functions/src/invitation.spec.ts:180]: please provide a status. Like this the test is falling for only one reason

File: apps/backend-functions/src/invitation.spec.ts [hunk 8]
```diff
@@ -153,4 +151,140 @@ describe('Invitation backend-function unit-tests', () => {
...
+    });
+
+    it('Invitation type is not \'attendEvent\', throws error', async () => {
+      const wrapped = testEnv.wrap(acceptOrDeclineInvitationAsAnonymous);
+
+      //Compose the call to simpleCallable cf with param data
+      const data: AnonymousInvitationAction = {
+        email: 'test@cascade8.com',
+        invitationId: 'I001',
+        status: null
```
Comment (phpgeekfr) [inline apps/backend-functions/src/invitation.spec.ts:202]: please provide a status. Like this the test is falling for only one reason

File: apps/backend-functions/src/invitation.spec.ts [hunk 9]
```diff
@@ -153,4 +151,140 @@ describe('Invitation backend-function unit-tests', () => {
...
+    });
+
+    it('Invitation mode is not \'invitation\', throws error', async () => {
+      const wrapped = testEnv.wrap(acceptOrDeclineInvitationAsAnonymous);
+
+      //Compose the call to simpleCallable cf with param data
+      const data: AnonymousInvitationAction = {
+        email: 'test@cascade8.com',
+        invitationId: 'I002',
+        status: null
```
Comment (phpgeekfr) [inline apps/backend-functions/src/invitation.spec.ts:224]: please provide a status. Like this the test is falling for only one reason

File: apps/backend-functions/src/invitation.spec.ts [hunk 10]
```diff
@@ -153,4 +151,140 @@ describe('Invitation backend-function unit-tests', () => {
...
+    });
+
+    it('Invitation to user email ID is not same, throws error', async () => {
+      const wrapped = testEnv.wrap(acceptOrDeclineInvitationAsAnonymous);
+
+      //Compose the call to simpleCallable cf with param data
+      const data: AnonymousInvitationAction = {
+        email: 'test1@cascade8.com',
+        invitationId: 'I003',
+        status: null
```
Comment (phpgeekfr) [inline apps/backend-functions/src/invitation.spec.ts:246]: please provide a status. Like this the test is falling for only one reason

File: apps/backend-functions/src/fixtures/inviteUsers.ts (hunk 1 above)
Comment (xmano) [reply apps/backend-functions/src/fixtures/inviteUsers.ts:32]: removed the consent related info.

File: apps/backend-functions/src/consent.ts (hunk 2 above)
Comment (xmano) [reply apps/backend-functions/src/consent.ts:54]: ok

File: apps/backend-functions/src/invitation.spec.ts (hunk 3 above)
Comment (xmano) [reply apps/backend-functions/src/invitation.spec.ts:66]: fixed this

File: apps/backend-functions/src/invitation.spec.ts (hunk 4 above)
Comment (xmano) [reply apps/backend-functions/src/invitation.spec.ts:160]: done

File: apps/backend-functions/src/invitation.spec.ts (hunk 6 above)
Comment (xmano) [reply apps/backend-functions/src/invitation.spec.ts:173]: Changed to I004

File: apps/backend-functions/src/invitation.spec.ts (hunk 7 above)
Comment (xmano) [reply apps/backend-functions/src/invitation.spec.ts:180]: status: 'pending'

File: apps/backend-functions/src/invitation.spec.ts (hunk 8 above)
Comment (xmano) [reply apps/backend-functions/src/invitation.spec.ts:202]: status: 'pending'

File: apps/backend-functions/src/invitation.spec.ts (hunk 10 above)
Comment (xmano) [reply apps/backend-functions/src/invitation.spec.ts:246]: status: 'pending'

File: apps/backend-functions/src/invitation.spec.ts (hunk 9 above)
Comment (xmano) [reply apps/backend-functions/src/invitation.spec.ts:224]: status: 'pending'

File: apps/backend-functions/src/fixtures/inviteUsers.ts (hunk 5 above)
Comment (xmano) [reply apps/backend-functions/src/fixtures/inviteUsers.ts:35]: removed

**Comment by xmano** [comment]
Fixed all review comments and ran locally

image

**Review by phpgeekfr — APPROVED** [review]

**Pull request merged**